### PR TITLE
Enable read from graphite using default path

### PR DESCRIPTION
**Pull request by Thib17**

### CODE REVIEW

File: graphite/http_helper.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
+// Copyright 2017 Thibault Chataigner <thibault.chataigner@gmail.com>
```
Comment (iksaif) [inline graphite/http_helper.go]: just name this file "read.go" or "http.go"

File: graphite/http_helper.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+		Scheme:     "http",
+		Host:       host,
+		Path:       path,
+		ForceQuery: true,
+		RawQuery:   values.Encode(),
+	}
+}
+
+func fetchUrl(u *url.URL, ctx context.Context) ([]byte, error) {
+	hresp, err := ctxhttp.Get(ctx, http.DefaultClient, u.String())
```
Comment (iksaif) [inline graphite/http_helper.go]: Add TODO to support basic auth + proxy

File: graphite/client.go [hunk 3]
```diff
@@ -123,44 +122,85 @@ func (c *Client) Write(samples model.Samples) error {
...
+	expandUrl := prepareUrl(c.graphite_web, expandEndpoint, map[string]string{"format": "json", "leavesOnly": "1", "query": queryStr})
+
+	// Get the list of targets
+	expandResponse := ExpandResponse{}
+	body, err := fetchUrl(expandUrl, ctx)
+	err = json.Unmarshal(body, &expandResponse)
 	if err != nil {
-		return nil, err
+		log.With("url", expandUrl).With("err", err).Warnln("Error parsing expand endpoint response body")
+		return targets
```
Comment (iksaif) [inline graphite/client.go]: return the error ?

File: graphite/client.go [hunk 4]
```diff
@@ -123,44 +122,85 @@ func (c *Client) Write(samples model.Samples) error {
...
-	// - query=<prefix>.<__name__>.**
+func (c *Client) handleReadQuery(query *remote.Query, ctx context.Context) *remote.QueryResult {
+	queryResult := &remote.QueryResult{}
+
+	from := strconv.Itoa(int(query.StartTimestampMs / 1000))
+	until := strconv.Itoa(int(query.EndTimestampMs / 1000))
+
+	targets := c.queryToTargets(query, ctx)
+	for _, target := range targets {
+		renderUrl := prepareUrl(c.graphite_web, renderEndpoint, map[string]string{"format": "json", "from": from, "until": until, "target": target})
```
Comment (iksaif) [inline graphite/client.go]: Can you put the inside of the loop in its own function ?

File: graphite/rules.go [hunk 5]
```diff
@@ -109,3 +115,19 @@ func defaultPath(m model.Metric, prefix string) string {
 	}
 	return buffer.String()
 }
+
+func metricLabelsFromPath(path string, prefix string) []*remote.LabelPair {
+	var labels []*remote.LabelPair
+	cleanedPath := strings.TrimPrefix(path, prefix)
```
Comment (iksaif) [inline graphite/rules.go]: Add a comment explaining that it uses the "default" export format to read back. (and quickly re-explain the format)

File: graphite/http_helper.go (hunk 1 above)
Comment (Thib17) [inline graphite/http_helper.go]: Done

File: graphite/http_helper.go (hunk 2 above)
Comment (Thib17) [inline graphite/http_helper.go]: Done

File: graphite/client.go (hunk 3 above)
Comment (Thib17) [inline graphite/client.go]: Done

File: graphite/client.go (hunk 4 above)
Comment (Thib17) [inline graphite/client.go]: Done

File: graphite/rules.go (hunk 5 above)
Comment (Thib17) [inline graphite/rules.go]: Done

**Pull request merged**